src/fmt.ml

```diff
@@ -65,6 +72,10 @@ let rec list ?(pp_sep = cut) pp_v ppf = function
 | v :: vs ->
     pp_v ppf v; if vs <> [] then (pp_sep ppf (); list ~pp_sep pp_v ppf vs)
 
+let hashtbl ?(pp_sep = cut) ~pp_k ~pp_v ppf tbl =
+  let f ppf k v = pf ppf "@[<1>%a:@.%a@]%a" pp_k k pp_v v pp_sep () in
```

Comment:
I don't see any flushing.

Having your `hashtbl_bindings` combinator is something I considered, yes, so it's fine by me.